scripts/lib/CIME/XML/env_batch.py

```diff
@@ -282,7 +282,7 @@ def set_job_defaults(self, batch_jobs, case):
 
                 walltime = self._default_walltime if walltime is None else walltime # last-chance fallback
 
-            self.set_value("JOB_QUEUE", queue, subgroup=job, ignore_type=specs is None)
```

Comment:
It totally does.   Specs is used here as an indication that the queue is defined in config_batch.xml, if it is not you need to use the ignore_type flag (which should be force option) to tell set value to use settings from the default queue to set up this new queue.   This was key to #2552 and I believe your changes will break what I just fixed.

Reply:
@jedwards4b , I'm still not seeing it. The variables "specs" is only defined if the block above (line 274) is entered. If it is entered, then the "walltime" variable is modified. I don't see how queue would get modified? The code as it was was not correct since I was getting "specs" variable undefined errors on melvin.